### DISCUSSION TITLE
How are messages stored in brokers in redis-smq?

### DISCUSSION THREAD

**Discussion by longtt2705** [weyoss/redis-smq#128] [Q&A] [answered]
In some other queue like Kafka, which is designed using `disk-based retention`, this means when a broker downs and then backs, the messages in queue can be recovered. But what happen in redis-smq, Redis relies on main memory of computer for data storage. Does it mean all the messages in the queue will be lost if a broker down? 
This is the thing that I concern to choose between redis-smq and others. Can you help me answer it @weyoss ?

**Comment by weyoss** [comment, marked as answer]
@longtt2705 RedisSMQ is backed by Redis. 

Depending on your application needs and requirements, Redis can be configured:

- To have no data persistence at all. When your Redis server shuts down, you lost your data.  
- To partially persist your data. Data is persisted into disk at specified intervals. 
- To have full data persistence "comparable to what PostgreSQL can provide you" quote. Every write operation is persisted.

For more details see https://redis.io/docs/manual/persistence
